app.js

```diff
@@ -34,37 +35,45 @@ const IDP_PATHS = {
   SIGN_OUT: '/signout',
   SETTINGS: '/settings'
 }
+const CERT_OPTIONS = [
+  'cert',
+  'key',
+  'encryptionCert',
+  'encryptionPublicKey',
+  'httpsPrivateKey',
+  'httpsCert',
+];
+const WILDCARD_ADDRESSES = ['0.0.0.0', '::'];
+const UNDEFINED_VALUE = 'None';
+const CRYPT_TYPES = {
+  certificate: /-----BEGIN CERTIFICATE-----[^-]*-----END CERTIFICATE-----/,
+  'RSA private key': /-----BEGIN RSA PRIVATE KEY-----\n[^-]*\n-----END RSA PRIVATE KEY-----/,
+  'public key': /-----BEGIN PUBLIC KEY-----\n[^-]*\n-----END PUBLIC KEY-----/,
+};
+const KEY_CERT_HELP_TEXT = dedent(chalk`
+  To generate a key/cert pair for the IdP, run the following command:
 
-const cryptTypes           = {
-      certificate: /-----BEGIN CERTIFICATE-----[^-]*-----END CERTIFICATE-----/,
-      'RSA private key': /-----BEGIN RSA PRIVATE KEY-----\n[^-]*\n-----END RSA PRIVATE KEY-----/,
-      'public key': /-----BEGIN PUBLIC KEY-----\n[^-]*\n-----END PUBLIC KEY-----/,
-    },
-    KEY_CERT_HELP_TEXT = "Please generate a key-pair for the IdP using the following openssl command:\n" +
-      "\topenssl req -x509 -new -newkey rsa:2048 -nodes -subj '/C=US/ST=California/L=San Francisco/O=JankyCo/CN=Test Identity Provider' -keyout idp-private-key.pem -out idp-public-cert.pem -days 7300";
-
+  {gray openssl req -x509 -new -newkey rsa:2048 -nodes \
+    -subj '/C=US/ST=California/L=San Francisco/O=JankyCo/CN=Test Identity Provider' \
+    -keyout idp-private-key.pem \
+    -out idp-public-cert.pem -days 7300}`
+);
 
 function matchesCertType(value, type) {
-  // console.info(`Testing ${cryptTypes[type].toString()} against "${value}"`);
-  // console.info(`result: ${cryptTypes[type] && cryptTypes[type].test(value)}`);
-  return cryptTypes[type] && cryptTypes[type].test(value);
-}
-
-function bufferFromString(value) {
-  if (Buffer.hasOwnProperty('from')) {
-    // node 6+
-    return Buffer.from(value);
-  } else {
-    return new Buffer(value);
-  }
+  return CRYPT_TYPES[type] && CRYPT_TYPES[type].test(value);
 }
 
 function resolveFilePath(filePath) {
+  if (filePath.startsWith('saml-idp/')) {
+    // Allows file path options to files included in this package, like config.js
+    const resolvedPath = require.resolve(filePath.replace(/^saml\-idp\//, `${__dirname}/`));
```

Comment:
the local repo name (e.g. `git clone`) can be something different than `saml-idp` which would be more constant for `npm install`.   I'm sure there is an environment variable for where the root is located so we are not hard-coding `saml-idp`.

Reply:
I'm actually using `__dirname` to resolve the file path here, which I think is what you're asking for. The primary purpose for this change is so that the default value for `--configFile` doesn't print as a full, resolved path. All paths should resolve just as they did before, regardless of wether the package is installed from npm or cloned using git.

You'll notice README.md contains an absolute path (`/Users/karl/src/saml-idp/config.js`) which appears to have come from the output of `node ./app --help` from Karl's src directory. With this change, `node ./app --help` will show the default value for `--configFile` as `saml-idp/config.js`.

As an added benefit, users can pass file paths to files in this package using the `saml-idp` root name. For example, `saml-idp --key saml-idp/idp-private-key.pem --cert  saml-idp/idp-public-cert.pem` will start the IDP using the packaged key / cert pair, something that wasn't previously possible if you had installed via `npm install saml-idp`. Or at least you would have had to specify the full path, something like `--key ./node_modules/saml-idp/idp-private-key.pem`.